### ISSUE TITLE
firefox syncserver fails to build 

### ISSUE THREAD

**Issue by mogorman** [NixOS/nixpkgs#29058]
## Issue description

I am trying to run firefox syncserver

### Steps to reproduce
in your configuration.nix add 

services.firefox.syncserver.enable = true;

## Technical details
my results 
```
nixos-rebuild switch  
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/550n7dy2nnkpgwf9wl12k6i07vfgfvjv-python2.7-testfixtures-4.5.0.drv
  /nix/store/7z8f3ib5g2vfgb7vnimkmpcfp6lpm9mw-python2.7-umemcache-1.6.3.drv
  /nix/store/lr0a36zxxwb7zkvx1kxqgbq0n92r5l75-python2.7-tokenlib-0.3.1.drv
  /nix/store/n60bc5qkkr9qpv2mmgbk1rg7vvlfvkdv-python2.7-pyramidhawkauth-0.1.0.drv
  /nix/store/hp666j37fsb2x3kr0bi1h0kzkqcz5lry-python2.7-serversyncstorage-1.5.13.drv
  /nix/store/znjqr317cwpa0wjnpkgwjk916ddf9bad-python2.7-tokenserver-1.2.11.drv
  /nix/store/6jidj1h7rh070i33cw9vpzqhy1fgdnxz-python2.7-syncserver-1.5.2.drv
  /nix/store/rckx0bisv64f98k9mlkvbyyjlmc0gl8q-unit-syncserver.service.drv
  /nix/store/x45mp4664aw3mlgp9wd51m76mlkbk7jg-system-units.drv
  /nix/store/2zna9vs8frdp49c7cizlmqma0rw7g5lz-etc.drv
  /nix/store/qmrq8jh2ccbfflgi633jk13iqjpdcqrq-nixos-system-marvin-17.03.1807.1360efeac6.drv
building path(s) ‘/nix/store/fzl0i0slglxha58h4arhyyhk110ysgh0-python2.7-testfixtures-4.5.0’
building path(s) ‘/nix/store/jjm3sxhnd2p7x5gzpggzw1v6kpbsvxd0-python2.7-tokenlib-0.3.1’
building path(s) ‘/nix/store/fs40fz95154zm80kdx5ykbmq3nwpd3mz-python2.7-umemcache-1.6.3’
unpacking sources
unpacking source archive /nix/store/cgr05ph8ld7v1bs1c6dxbxj10i5j0xsw-umemcache-1.6.3.zip
source root is umemcache-1.6.3
setting SOURCE_DATE_EPOCH to timestamp 1370870218 of file umemcache-1.6.3/PKG-INFO
patching sources
configuring
unpacking sources
unpacking source archive /nix/store/ayc8f40d4k1kyk9isb020qrvqsfyrl8x-tokenlib
building
source root is tokenlib
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file tokenlib/tox.ini
patching sources
configuring
building
unpacking sources
unpacking source archive /nix/store/wm6chw36pgp16psjxi4fy7cb9bmzv7v0-testfixtures-4.5.0.tar.gz
source root is testfixtures-4.5.0
setting SOURCE_DATE_EPOCH to timestamp 1447398962 of file testfixtures-4.5.0/testfixtures.egg-info/top_level.txt
patching sources
configuring
building
running bdist_wheel
running build
running build_ext
building 'umemcache' extension
creating build
creating build/temp.linux-x86_64-2.7
creating build/temp.linux-x86_64-2.7/python
creating build/temp.linux-x86_64-2.7/lib
gcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWIN32_LEAN_AND_MEAN -I./lib/ -I/nix/store/7dxqxqc68xfhip2rzi6vqn552f0f7phk-python-2.7.13/include/python2.7 -c ./python/umemcache.cpp -o build/temp.linux-x86_64-2.7/./python/umemcache.o
cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/tokenlib
copying tokenlib/__init__.py -> build/lib/tokenlib
copying tokenlib/utils.py -> build/lib/tokenlib
copying tokenlib/errors.py -> build/lib/tokenlib
creating build/lib/tokenlib/tests
copying tokenlib/tests/__init__.py -> build/lib/tokenlib/tests
copying tokenlib/tests/test_utils.py -> build/lib/tokenlib/tests
copying tokenlib/tests/test_tokens.py -> build/lib/tokenlib/tests
running egg_info
creating tokenlib.egg-info
writing tokenlib.egg-info/PKG-INFO
writing top-level names to tokenlib.egg-info/top_level.txt
writing dependency_links to tokenlib.egg-info/dependency_links.txt
writing manifest file 'tokenlib.egg-info/SOURCES.txt'
reading manifest file 'tokenlib.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'tokenlib.egg-info/SOURCES.txt'
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/tokenlib
copying build/lib/tokenlib/__init__.py -> build/bdist.linux-x86_64/wheel/tokenlib
copying build/lib/tokenlib/utils.py -> build/bdist.linux-x86_64/wheel/tokenlib
copying build/lib/tokenlib/errors.py -> build/bdist.linux-x86_64/wheel/tokenlib
creating build/bdist.linux-x86_64/wheel/tokenlib/tests
copying build/lib/tokenlib/tests/__init__.py -> build/bdist.linux-x86_64/wheel/tokenlib/tests
copying build/lib/tokenlib/tests/test_utils.py -> build/bdist.linux-x86_64/wheel/tokenlib/tests
copying build/lib/tokenlib/tests/test_tokens.py -> build/bdist.linux-x86_64/wheel/tokenlib/tests
running install_egg_info
Copying tokenlib.egg-info to build/bdist.linux-x86_64/wheel/tokenlib-0.3.1-py2.7.egg-info
running install_scripts
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/testfixtures
copying testfixtures/comparison.py -> build/lib/testfixtures
copying testfixtures/__init__.py -> build/lib/testfixtures
copying testfixtures/utils.py -> build/lib/testfixtures
copying testfixtures/manuel.py -> build/lib/testfixtures
copying testfixtures/shouldraise.py -> build/lib/testfixtures
copying testfixtures/outputcapture.py -> build/lib/testfixtures
copying testfixtures/tempdirectory.py -> build/lib/testfixtures
copying testfixtures/components.py -> build/lib/testfixtures
copying testfixtures/popen.py -> build/lib/testfixtures
copying testfixtures/replace.py -> build/lib/testfixtures
copying testfixtures/tdatetime.py -> build/lib/testfixtures
copying testfixtures/logcapture.py -> build/lib/testfixtures
copying testfixtures/rmtree.py -> build/lib/testfixtures
copying testfixtures/resolve.py -> build/lib/testfixtures
copying testfixtures/compat.py -> build/lib/testfixtures
creating build/bdist.linux-x86_64/wheel/tokenlib-0.3.1.dist-info/WHEEL
creating build/lib/testfixtures/tests
copying testfixtures/tests/__init__.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_datetime.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_replacer.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_roundcomparison.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_popen.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_wrap.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_docs.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_log_capture.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_comparison.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_manuel.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_popen_docs.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_generator.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_tempdir.py -> build/lib/testfixtures/tests
copying testfixtures/tests/sample2.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_components.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_diff.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_manuel_examples.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_date.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_time.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_should_raise.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_replace.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_stringcomparison.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_compare.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_tempdirectory.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_logcapture.py -> build/lib/testfixtures/tests
copying testfixtures/tests/test_outputcapture.py -> build/lib/testfixtures/tests
copying testfixtures/tests/compat.py -> build/lib/testfixtures/tests
copying testfixtures/tests/sample1.py -> build/lib/testfixtures/tests
running egg_info
installing
/tmp/nix-build-python2.7-tokenlib-0.3.1.drv-0/tokenlib/dist /tmp/nix-build-python2.7-tokenlib-0.3.1.drv-0/tokenlib
writing requirements to testfixtures.egg-info/requires.txt
writing testfixtures.egg-info/PKG-INFO
writing top-level names to testfixtures.egg-info/top_level.txt
writing dependency_links to testfixtures.egg-info/dependency_links.txt
reading manifest file 'testfixtures.egg-info/SOURCES.txt'
writing manifest file 'testfixtures.egg-info/SOURCES.txt'
copying testfixtures/version.txt -> build/lib/testfixtures
copying testfixtures/tests/configparser-read.txt -> build/lib/testfixtures/tests
copying testfixtures/tests/configparser-write.txt -> build/lib/testfixtures/tests
copying testfixtures/tests/directory-contents.txt -> build/lib/testfixtures/tests
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/comparison.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/__init__.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/utils.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/manuel.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/version.txt -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/shouldraise.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/outputcapture.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/tempdirectory.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/components.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/popen.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/replace.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/tdatetime.py -> build/bdist.linux-x86_64/wheel/testfixtures
creating build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/__init__.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_datetime.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_replacer.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_roundcomparison.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_popen.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/configparser-write.txt -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_wrap.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_docs.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_log_capture.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_comparison.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_manuel.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_popen_docs.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_generator.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_tempdir.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/sample2.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_components.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_diff.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_manuel_examples.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_date.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_time.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_should_raise.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_replace.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_stringcomparison.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_compare.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_tempdirectory.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/configparser-read.txt -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_logcapture.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/test_outputcapture.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/compat.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/directory-contents.txt -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/tests/sample1.py -> build/bdist.linux-x86_64/wheel/testfixtures/tests
copying build/lib/testfixtures/logcapture.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/rmtree.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/resolve.py -> build/bdist.linux-x86_64/wheel/testfixtures
copying build/lib/testfixtures/compat.py -> build/bdist.linux-x86_64/wheel/testfixtures
running install_egg_info
Copying testfixtures.egg-info to build/bdist.linux-x86_64/wheel/testfixtures-4.5.0-py2.7.egg-info
running install_scripts
./python/umemcache.cpp: In function 'int Client_init(PyClient*, PyObject*, PyObject*)':
./python/umemcache.cpp:255:60: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
   static char *kwlist[] = {"address", "max_item_size", NULL};
                                                            ^
./python/umemcache.cpp:255:60: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp: In function 'PyObject* Client_incr(PyClient*, PyObject*)':
./python/umemcache.cpp:822:62: error: format not a string literal and no format arguments [-Werror=format-security]
         return PyErr_Format(umemcache_MemcachedError, pResult);
                                                              ^
./python/umemcache.cpp: In function 'PyObject* Client_decr(PyClient*, PyObject*)':
./python/umemcache.cpp:870:62: error: format not a string literal and no format arguments [-Werror=format-security]
         return PyErr_Format(umemcache_MemcachedError, pResult);
                                                              ^
./python/umemcache.cpp: At global scope:
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
 };
 ^
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp:1056:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
./python/umemcache.cpp: In function 'void initumemcache()':
./python/umemcache.cpp:1124:31: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
       PyExc_RuntimeError, NULL); 
                               ^
creating build/bdist.linux-x86_64/wheel/testfixtures-4.5.0.dist-info/WHEEL
installing
/tmp/nix-build-python2.7-testfixtures-4.5.0.drv-0/testfixtures-4.5.0/dist /tmp/nix-build-python2.7-testfixtures-4.5.0.drv-0/testfixtures-4.5.0
cc1plus: some warnings being treated as errors
error: command 'gcc' failed with exit status 1
builder for ‘/nix/store/7z8f3ib5g2vfgb7vnimkmpcfp6lpm9mw-python2.7-umemcache-1.6.3.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/hp666j37fsb2x3kr0bi1h0kzkqcz5lry-python2.7-serversyncstorage-1.5.13.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/6jidj1h7rh070i33cw9vpzqhy1fgdnxz-python2.7-syncserver-1.5.2.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/rckx0bisv64f98k9mlkvbyyjlmc0gl8q-unit-syncserver.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/x45mp4664aw3mlgp9wd51m76mlkbk7jg-system-units.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/2zna9vs8frdp49c7cizlmqma0rw7g5lz-etc.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/qmrq8jh2ccbfflgi633jk13iqjpdcqrq-nixos-system-marvin-17.03.1807.1360efeac6.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/qmrq8jh2ccbfflgi633jk13iqjpdcqrq-nixos-system-marvin-17.03.1807.1360efeac6.drv’ failed
```
```
# nixos-version 
17.03.1807.1360efeac6 (Gorilla)
# nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"17.03.1807.1360efeac6"
# grep build-use-sandbox /etc/nix/nix.conf
build-use-sandbox = false
```

**Comment by mogorman** [comment]
im guesing it was never backported now. as i dont have these fixes in 15.03

**Comment by mogorman** [comment]
This is fixed in 15.09 so i see no reason to leave this open any longer.
